src/query/server/query.go

```diff
@@ -743,6 +747,12 @@ func newDownsampler(
 		return nil, errors.Wrap(err, "unable to create KV store from the "+
 			"cluster management config client")
 	}
+	if applyCustomRuleStore != nil {
+		kvStore, err = applyCustomRuleStore(kvStore)
```

Comment:
Do you need to return the kvStore here if you're passing it in?

Reply:
yeah, idea is the aggregator will use the overridden store that's returned back